GetDKAN/dkan
# Remove distribution records from DB tables when a dataset is deleted.

fixes [#3343]

![EventFlow](https://user-images.githubusercontent.com/314172/119897128-975b0b00-bf05-11eb-8e02-013566c876fc.png)


View information about a dataset:
`dktl dkan:dataset-info {id}`

## QA Steps
If possible start with an empty database.
- [ ] Add a dataset with a csv distribution
      
```
      dktl drush dkan:harvest:register '{"identifier":"h1","extract":{"type":"\\Harvest\\ETL\\Extract\\DataJson","uri":"https://dkan-default-content-files.s3.amazonaws.com/data.json"},"transforms":[],"load":{"type":"\\Drupal\\harvest\\Load\\Dataset"}}'
      
      dktl drush dkan:harvest:run h1
```
- [ ] `drush queue:run datastore_import`
- [ ] Look at the database, you should see:

      - a datastore table
      - 3 records in the _dkan_metastore_resource_mapper_ table (source, local_file, local_url)
      - 1 record in the _jobstore_dkan_datastore_importer_ table
      - 1 record in the _jobstore_filefetcher_filefetcher_ table
      
- [ ] Delete the dataset you just made `dktl drush dkan:harvest:revert h1`
- [ ] `drush queue:run orphan_reference_processor`

Confirm the database has been "cleaned":
- [ ] The datastore should be dropped. 
- [ ] The 3 dkan_metastore_resource_mapper records should be removed, 
- [ ] jobstore_filefetcher_filefetcher is empty, 
- [ ] jobstore_datastore_import is empty,
- [ ] The file and directory of the resource should be gone: _sites/files/resources/{ref_uuid}/file.csv_
